src/main/java/com/adobe/target/edge/client/ondevice/collator/CustomParamsCollator.java

```diff
@@ -35,4 +36,28 @@ private void addAllParameters(Map<String, Object> custom, RequestDetails details
       }
     }
   }
+
+  private Map<String, Object> createNestedParametersFromDots(Map<String, Object> custom) {
+    Map<String, Object> result = new HashMap<>();
+    custom.forEach(
+        (key, value) -> {
+          if (key.contains(".")
+              && !key.contains("..")
+              && key.charAt(0) != '.'
+              && key.charAt(key.length() - 1) != '.') {
```

Comment:
could you clarify by what you mean by filter out blanks? As in don't process keys that have blanks after split? Or remove strings from the split array that are blank? For example: would `{"this..should..be": "ignored"}` stay as is or be transformed to: `{"this" : { "should" : { "be": "ignored"}}}`

Reply:
yeah i was meaning to filter out blanks in the array after the split.  but on second thought i think it will not work unless we change rule generation on smokescreen as well.  so i think your implementation is 👍